File: languages/go/entity/identity/boolean_attribute_descriptor.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// BooleanAttributeDescriptor represents a descriptor for a boolean attribute.
type BooleanAttributeDescriptor struct {
	AttributeDescriptor
}

// NewBooleanAttributeDescriptor creates a new BooleanAttributeDescriptor instance with the provided details.
func NewBooleanAttributeDescriptor(name string, id string, description string, required bool) BooleanAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return BooleanAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func NewBooleanAttributeDescriptorFromProto(s *proto.BooleanAttributeDefinition) BooleanAttributeDescriptor {
	if s == nil {
		return BooleanAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return BooleanAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func (s BooleanAttributeDescriptor) ToProto() *proto.BooleanAttributeDefinition {
	return &proto.BooleanAttributeDefinition{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
}

File: languages/js/src/entity/identity/boolean_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/bloock_identity_entities";
import { AttributeDescriptor } from "./attribute_descriptor";

/**
 * Represents a descriptor for a boolean attribute.
 */
export class BooleanAttributeDescriptor extends AttributeDescriptor {
  public toProto(): identityEntitiesProto.BooleanAttributeDefinition {
    return identityEntitiesProto.BooleanAttributeDefinition.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required
    });
  }

  static fromProto(
    r: identityEntitiesProto.BooleanAttributeDefinition
  ): BooleanAttributeDescriptor {
    return new BooleanAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required
    );
  }
}
